tests/transports/test_all_plugins.py

```diff
@@ -710,251 +608,238 @@ def test_put_and_get_overwrite(
-        file_3 = os.path.join(local_base_dir, 'c.txt')
+        file_1 = local_base_dir / 'a.txt'
+        file_2 = local_base_dir / 'b.tmp'
+        file_3 = local_base_dir / 'c.txt'
         text = 'Viva Verdi\n'
         for filename in [file_1, file_2, file_3]:
             with open(filename, 'w', encoding='utf8') as fhandle:
                 fhandle.write(text)
 
-        # first test putransport. Copy of two files matching patterns, into a folder
-        transport.put(os.path.join(local_base_dir, '*.txt'), '.')
-        assert set(['a.txt', 'c.txt', 'local']) == set(transport.listdir('.'))
-        transport.remove('a.txt')
-        transport.remove('c.txt')
-        # second. Copy of folder into a non existing folder
-        transport.put(local_base_dir, 'prova')
-        assert set(['prova', 'local']) == set(transport.listdir('.'))
-        assert set(['a.txt', 'b.tmp', 'c.txt']) == set(transport.listdir('prova'))
-        transport.rmtree('prova')
-        # third. copy of folder into an existing folder
-        transport.mkdir('prova')
-        transport.put(local_base_dir, 'prova')
-        assert set(['prova', 'local']) == set(transport.listdir('.'))
-        assert set(['local']) == set(transport.listdir('prova'))
-        assert set(['a.txt', 'b.tmp', 'c.txt']) == set(transport.listdir(os.path.join('prova', 'local')))
-        transport.rmtree('prova')
-        # third test copy. Can copy one file into a new file
-        transport.put(os.path.join(local_base_dir, '*.tmp'), 'prova')
-        assert set(['prova', 'local']) == set(transport.listdir('.'))
-        transport.remove('prova')
-        # fourth test copy: can't copy more than one file on the same file,
+        # first test the put. Copy of two files matching patterns, into a folder
+        transport.put(str(local_base_dir / '*.txt'), str(remote_workdir))
+        assert set(['a.txt', 'c.txt']) == set(transport.listdir(str(remote_workdir)))
+        transport.remove(str(remote_workdir / 'a.txt'))
+        transport.remove(str(remote_workdir / 'c.txt'))
+
+        # second test put. Put of two folders
+        transport.put(str(local_base_dir), str(remote_workdir / 'prova'))
+        assert set(['prova']) == set(transport.listdir(str(remote_workdir)))
+        assert set(['a.txt', 'b.tmp', 'c.txt']) == set(transport.listdir(str(remote_workdir / 'prova')))
+        transport.rmtree(str(remote_workdir / 'prova'))
+
+        # third test put. Can copy one file into a new file
+        transport.put(str(local_base_dir / '*.tmp'), str(remote_workdir / 'prova'))
+        assert transport.isfile(str(remote_workdir / 'prova'))
+        transport.remove(str(remote_workdir / 'prova'))
+
+        # fourth test put: can't copy more than one file on the same file,
         # i.e., the destination should be a folder
         with pytest.raises(OSError):
-            transport.put(os.path.join(local_base_dir, '*.txt'), 'prova')
-        # copy of folder into file
```

Comment:
not clear why these lines are removed?
```python
        # copy of folder into file

        with open(os.path.join(local_dir, directory, 'existing.txt'), 'w', encoding='utf8') as fhandle:
            fhandle.write(text)
        with pytest.raises(OSError):
            transport.put(os.path.join(local_base_dir), 'existing.txt')
        transport.remove('existing.txt')
```

Reply:
Delete by mistake, I add it back as:

```python
        # can't copy folder to an exist file
        with open(remote_workdir / 'existing.txt', 'w', encoding='utf8') as fhandle:
            fhandle.write(text)
        with pytest.raises(OSError):
            transport.put(str(local_base_dir), str(remote_workdir / 'existing.txt'))
        transport.remove(str(remote_workdir / 'existing.txt'))
```

Nice catch!!